mlir/lib/Conversion/TosaToMIOpen/TosaToMIOpen.cpp

```diff
@@ -78,32 +78,9 @@ static Value expandMemRef(ConversionPatternRewriter &rw, Operation *op,
   }
   ArrayRef<int64_t> shape = oprType.getShape();
 
-  SmallVector<SmallString<8>> dimNames;
-  SmallVector<StringRef> dimNameRefs;
-  dimNames.reserve(shape.size());
-  dimNameRefs.reserve(shape.size());
-  for (size_t i = 0, e = shape.size(); i < e; ++i) {
-    SmallString<8> dimName;
-    ("dim" + Twine(i)).toVector(dimName);
-    dimNames.emplace_back(std::move(dimName));
-    dimNameRefs.push_back(StringRef(dimNames[i]));
-  }
-
-  miopen::BottomUpCTBuilder transform(rw, dimNameRefs, shape, loc);
-  llvm::StringMap<uint32_t> upperNames;
-  if (idx == 0) {
-    StringRef expandedDim = dimNameRefs[0];
-    upperNames = miopen::expandNamesInPlace(
-        dimNameRefs, {{expandedDim, {"g", expandedDim}}});
-  } else {
-    StringRef expandedDim = dimNameRefs[idx - 1];
-    upperNames = miopen::expandNamesInPlace(
-        dimNameRefs, {{expandedDim, {expandedDim, "g"}}});
-  }
+  miopen::BottomUpCTBuilder transform(rw, shape, loc);
+  transform.expand({idx}, {1});
 
-  miopen::BottomUpCTTopDimsWrapper wrapper(transform, upperNames);
-  wrapper.passThrough(dimNameRefs);
```

Comment:
And so this'll be (not needing the wrapper) `.passThrough({0, 1, ... idx - 1, idx + 1, ... n + 1}, {0, 1, ... n})` and then `addDim("g", idx, size)`